examples/complete-example/main.tf

```diff
@@ -2,61 +2,73 @@
 # CREATE AN EXAMPLE VPC WITH SUBNETS, INTERNET- AND NAT GATEWAYS
 # ---------------------------------------------------------------------------------------------------------------------
 
-provider "aws" {
-  region = var.aws_region
+locals {
+  cidr_block = "10.0.0.0/16"
 }
 
 module "vpc" {
   source = "../.."
 
   module_enabled = true
 
-  vpc_name            = "Test"
-  cidr_block          = "10.0.0.0/16"
-  create_nat_gateways = "one_per_az"
+  vpc_name   = "main"
+  cidr_block = local.cidr_block
+  # nat_gateway_mode = "one_per_az" # (defaults to "single")
+  # nat_gateway_mode = "none" # (defaults to "single")
 
-  public_subnets = [
-    {
-      availability_zone = "us-east-1a",
-      cidr_block        = "10.0.64.0/21",
-    },
-    {
-      availability_zone = "us-east-1a",
-      cidr_block        = "10.0.80.0/21",
-    },
+  subnets = [
     {
-      availability_zone = "us-east-1b",
-      cidr_block        = "10.0.96.0/21",
-    }
-  ]
+      group = "main"   # (default: default)
+      class = "public" # (default: private)
 
-  private_subnets = [
-    {
-      availability_zone = "us-east-1a",
-      cidr_block        = "10.0.112.0/21",
+      map_public_ip_on_launch         = true
+      assign_ipv6_address_on_creation = false
+
+      cidr_block = cidrsubnet(local.cidr_block, 4, 0) # (default: cidr_block of vpc)
+      newbits    = 4                                  # (default: 8)
+      netnums_by_az = {
+        a = [0]
+        b = [1]
+      }
+      routes = [
+      ]
+      tags = {
+      }
```

Comment:
Could you add documentation? Our goal is to add lean and easy-to-understand API's. Even after looking at this code block for 2 minutes, it isn't clear to me what this is doing.

Reply:
what exactly is unclear?

Reply:
added more context... @soerenmartius

Reply:
https://github.com/mineiros-io/terraform-aws-vpc/blob/2580355b89632bf66ef2d21983c60a39b2aad3e0/examples/complete-example/main.tf